### PR TITLE
Use mainnet permit2

### PR DESCRIPTION
**Pull request by hensha256**

### CODE REVIEW

File: contracts/modules/PaymentsImmutables.sol [hunk 1]
```diff
@@ -2,7 +2,7 @@
 pragma solidity ^0.8.24;
 
 import {IWETH9} from '../interfaces/external/IWETH9.sol';
-import {IAllowanceTransfer} from 'permit2/src/interfaces/IAllowanceTransfer.sol';
+import {IPermit2} from 'permit2/src/interfaces/IPermit2.sol';
```
Comment (ewilz) [inline contracts/modules/PaymentsImmutables.sol:5]: just wondering why. readability?

File: hardhat.config.ts [hunk 2]
```diff
@@ -45,7 +30,7 @@ export default {
       chainId: 1,
       forking: {
         url: `https://mainnet.infura.io/v3/${process.env.INFURA_API_KEY}`,
-        blockNumber: 15360000,
+        blockNumber: 20010000,
```
[the remark this answers is not in the document]
Comment (ewilz) [reply hardhat.config.ts:33]: responding to my own post. Apparently not pinning a block results in slower tests bc it's not cached. So probably fine

File: test/integration-tests/gas-tests/__snapshots__/Uniswap.gas.test.ts.snap [hunk 3]
```diff
@@ -3,105 +3,105 @@
...
-  "gasUsed": 187260,
+  "gasUsed": 186913,
 }
 `;
 
 exports[`Uniswap Gas Tests Mixing V2 and V3 with Universal Router. Split routes gas: ETH --> ERC20 split V2 and V3, one hop 1`] = `
 Object {
   "calldataByteLength": 1124,
-  "gasUsed": 199851,
+  "gasUsed": 202553,
```
Comment (ewilz) [inline test/integration-tests/gas-tests/__snapshots__/Uniswap.gas.test.ts.snap:104]: hmm

File: test/integration-tests/shared/mainnetForkHelpers.ts [hunk 4]
```diff
@@ -102,14 +104,14 @@ export const executeSwapRouter02Swap = async (
...
 
-export const resetFork = async (block: number = 15360000) => {
+export const resetFork = async (block: number = 20010000) => {
   await hre.network.provider.request({
     method: 'hardhat_reset',
     params: [
       {
         forking: {
           jsonRpcUrl: `https://mainnet.infura.io/v3/${process.env.INFURA_API_KEY}`,
-          blockNumber: block,
```
Comment (ewilz) [inline test/integration-tests/shared/mainnetForkHelpers.ts:112]: why take a parameter if it doesn't get used?

Comment (ewilz) [reply test/integration-tests/shared/mainnetForkHelpers.ts:112]: or... also do we even need to reset the fork anymore if we're always using the same block number?

**Review by ewilz — APPROVED** [review]
looks good....left some nits as well as wondering about some gas weirdness

File: test/integration-tests/gas-tests/__snapshots__/Uniswap.gas.test.ts.snap (hunk 3 above)
Comment (hensha256) [reply test/integration-tests/gas-tests/__snapshots__/Uniswap.gas.test.ts.snap:104]: oh interesting

File: test/integration-tests/shared/mainnetForkHelpers.ts (hunk 4 above)
Comment (hensha256) [reply test/integration-tests/shared/mainnetForkHelpers.ts:112]: oh this was from some debugging, will remove the parameter given we are aligning to 1 block

File: contracts/modules/PaymentsImmutables.sol (hunk 1 above)
Comment (hensha256) [reply contracts/modules/PaymentsImmutables.sol:5]: oh we do need IPermit2 to compile so we have it in typechain - so keeping it

File: test/integration-tests/gas-tests/__snapshots__/UniversalVSSwapRouter.gas.test.ts.snap [hunk 5]
```diff
@@ -3,36 +3,36 @@
...
+exports[`Uniswap UX Tests gas: Comparisons Frequent Swapper - 10 swaps Permit2 Max Approval Swap 1`] = `4374775`;
 
-exports[`Uniswap UX Tests gas: Comparisons Frequent Swapper - 10 swaps Permit2 Sign Per Swap 1`] = `3384615`;
+exports[`Uniswap UX Tests gas: Comparisons Frequent Swapper - 10 swaps Permit2 Sign Per Swap 1`] = `4528409`;
 
-exports[`Uniswap UX Tests gas: Comparisons Frequent Swapper - 10 swaps SwapRouter02 1`] = `3320620`;
+exports[`Uniswap UX Tests gas: Comparisons Frequent Swapper - 10 swaps SwapRouter02 1`] = `3222565`;
 
-exports[`Uniswap UX Tests gas: Comparisons Frequent Swapper across 3 swap router versions - 15 swaps across 3 versions Permit2 Max Approval Swap 1`] = `4317753`;
+exports[`Uniswap UX Tests gas: Comparisons Frequent Swapper across 3 swap router versions - 15 swaps across 3 versions Permit2 Max Approval Swap 1`] = `5466861`;
```
[the remark this answers is not in the document]
Comment (hensha256) [reply test/integration-tests/gas-tests/__snapshots__/UniversalVSSwapRouter.gas.test.ts.snap]: ok fixed!! was just using bad fee tiers - ive made it so the fee tiers for all v3 pools are looked up in 1 function so its uniform

**Pull request merged**